server/cluster/cluster.go

```diff
@@ -1259,7 +1287,40 @@ func (c *RaftCluster) UpStore(storeID uint64) error {
 		zap.String("store-address", newStore.GetAddress()))
 	err := c.putStoreLocked(newStore)
 	if err == nil {
-		c.resetRemovingProgress(storeID, store.GetAddress(), removingAction)
+		c.resetProgress(storeID, store.GetAddress(), removingAction)
+	}
+	return err
+}
+
+// ReadyToServe change store's node state to Serving.
+func (c *RaftCluster) ReadyToServe(storeID uint64) error {
```

Comment:
Could you please merge it with the function `UpStore`, it seems the logic is similar except for the `options`?

Reply:
These two functions are a little bit different, the common part is only checking the store state.